src/engine/Match_dependency.ml

```diff
@@ -51,9 +51,9 @@ let match_dependency_formula :
     Lockfile_target.t ->
     Rule.dependency_formula ->
     Pattern_match.dependency_match list =
- fun { lazy_lockfile_ast_and_errors; _ } ->
+ fun { lazy_ast_and_errors; _ } ->
   List.concat_map (fun pat ->
-      match_dependency_pattern (Lazy.force lazy_lockfile_ast_and_errors) pat)
+      match_dependency_pattern (Lazy.force lazy_ast_and_errors) pat)
```

Comment:
I'd go easy on the lazy computations. Is the laziness really needed? It makes things harder to debug imo.

Reply:
I believe we currently do rely on the AST being lazy in some cases (see the comment in what is now `Xtarget.parse_file`, at least for `Xtarget.t`, not `Lockfile_target.t`). I would defer to @mmcqd who added this in https://github.com/semgrep/semgrep/pull/9695.

Reply:
I just copied the way that `Xtarget` works, I don't have any thought out need for laziness actually

Reply:
I am resolving this thread here since it is not related to this PR; if changes are desired Matthew can do so in a separate PR.